sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java

```diff
@@ -53,105 +53,73 @@
+    if (!reports.isEmpty()) {
+      LOG.info("Python test coverage");
+      for (File report : reports) {
+        Map<InputFile, NewCoverage> coverageMeasures = parseReport(report, context);
+        saveMeasures(coverageMeasures, filesCovered);
+      }
+    }
     if (settings.getBoolean(FORCE_ZERO_COVERAGE_KEY)) {
       LOG.debug("Zeroing coverage information for untouched files");
-      zeroMeasuresWithoutReports(context, filesCoveredByUT, filesCoveredByIT, filesCoveredOverall, linesOfCode);
+      zeroMeasuresWithoutReports(context, filesCovered, linesOfCode);
     }
   }
 
-  private static void zeroMeasuresWithoutReports(
-    SensorContext context,
-    HashSet<InputFile> filesCoveredByUT,
-    HashSet<InputFile> filesCoveredByIT,
-    HashSet<InputFile> filesCoveredOverall,
-    Map<InputFile, Set<Integer>> linesOfCode
-  ) {
+  private static void zeroMeasuresWithoutReports(SensorContext context, HashSet<InputFile> filesCovered, Map<InputFile, Set<Integer>> linesOfCode) {
     FileSystem fileSystem = context.fileSystem();
     FilePredicates p = fileSystem.predicates();
     Iterable<InputFile> inputFiles = fileSystem.inputFiles(p.and(p.hasType(InputFile.Type.MAIN), p.hasLanguage(Python.KEY)));
     for (InputFile inputFile : inputFiles) {
-      Set<Integer> linesOfCodeForFile = linesOfCode.get(inputFile);
-
-      if (!filesCoveredByUT.contains(inputFile)) {
-        saveZeroValueForResource(inputFile, context, CoverageType.UNIT, linesOfCodeForFile);
-      }
-
-      if (!filesCoveredByIT.contains(inputFile)) {
-        saveZeroValueForResource(inputFile, context, CoverageType.IT, linesOfCodeForFile);
-      }
-
-      if (!filesCoveredOverall.contains(inputFile)) {
-        saveZeroValueForResource(inputFile, context, CoverageType.OVERALL, linesOfCodeForFile);
+      if (!filesCovered.contains(inputFile)) {
+        saveZeroValueForResource(inputFile, context, linesOfCode.get(inputFile));
       }
     }
   }
 
-  private static void saveZeroValueForResource(InputFile inputFile, SensorContext context, CoverageType ctype, @Nullable Set<Integer> linesOfCode) {
+  private static void saveZeroValueForResource(InputFile inputFile, SensorContext context, @Nullable Set<Integer> linesOfCode) {
     if (linesOfCode != null) {
       if (LOG.isDebugEnabled()) {
-        LOG.debug("Zeroing {} coverage measures for file '{}'", ctype, inputFile.relativePath());
+        LOG.debug("Zeroing coverage measures for file '{}'", inputFile.relativePath());
       }
-
       NewCoverage newCoverage = context.newCoverage()
-        .onFile(inputFile)
-        .ofType(ctype);
+        .onFile(inputFile);
       linesOfCode.forEach((Integer line) -> newCoverage.lineHits(line, 0));
       newCoverage.save();
     }
   }
 
-
-  private Map<InputFile, NewCoverage> parseReports(List<File> reports, SensorContext context) {
+  private Map<InputFile, NewCoverage> parseReport(File report, SensorContext context) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "parseReport" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.python%3Apython&branch=SONARPY-227&issues=AWDq-kQSvOZGGKctVrGW&open=AWDq-kQSvOZGGKctVrGW)

[ISSUE_KEY]: AWDq-kQSvOZGGKctVrGW